packages/fuels-abigen-macro/tests/harness.rs

```diff
@@ -1545,3 +1545,30 @@ async fn unit_type_enums() {
         .unwrap();
     assert_eq!(result.value, BimBamBoum::Boum());
 }
+
+#[tokio::test]
+// TODO(vnepveu): test for multiple assets
+async fn test_provider_balance_api() {
+    let mut wallet = LocalWallet::new_random(None);
+    let coins = setup_coins(wallet.address(), 21, 11);
+    let (provider, _) = setup_test_provider(coins.clone(), Config::local_node()).await;
+    wallet.set_provider(provider);
+    for (_utxo_id, coin) in coins {
+        let balance = wallet
+            .get_provider()
+            .unwrap()
+            .get_asset_balance(&wallet.address(), coin.asset_id)
```

Comment:
Very unergonomic to do `wallet.<...>.get_asset_balance(&wallet.address, asset_id)`. Why do I have to pass a reference to my own wallet? I'm using my own `wallet` instance and having to reference itself on its own method. Weird.

Make it `wallet.<...>.get_asset_balance(asset_id)`. If you want to query other wallets' balances, use a provider directly (and make sure `Provider` exposes a `get_asset_balance(address, asset_id)` as well).